charts/consul/templates/api-gateway-controller-deployment.yaml

```diff
@@ -119,6 +119,10 @@ spec:
           value: {{ .Values.global.adminPartitions.name }}
         {{- end }}
         {{- end }}
+        {{- if not .Values.client.enabled }}
```

Comment:
If clients are enabled we still leverage `$(HOST_IP)` -- so in any other case (i.e. cluster-based or external servers) we know we should use the server gRPC discovery components. This is also passed explicitly as an env variable to support older versions of api-gateway rather than having the entrypoints fail with an unknown flag.